src/components/xy_chart/crosshairs/react_vis_crosshair_y.js

```diff
@@ -0,0 +1,170 @@
+// TODO this is a more or less a copy and paste of crosshair y
```

Comment:
what's the purpose for having the two distinct files?

Reply:
Maybe I wrote it in a wrong way: the `react_vis_crosshair_y.js`is the axis-inverted version of the crosshair component of `react-vis`. Than on top of the Crosshair component I've build our EuiCrosshairX and Y that doesn't extend Crosshair but AbstractSeries so we can get the mousemove handlers.

Reply:
BTW I wasn't sure what can be better here. Keep them as two separate components, or try to build an abstract one.
Also I'm not sure how to handle the license code, since the `react_vis_crosshair_y.js` file is a copy end paste from the react-vis lib with slight modified code.

Reply:
The preference of this implementation is a bit slow it seems compared to what @chandlerprall and I built. With extreme sizes of data sets, this implementation lags a fair bit on my machine...

Reply:
@mattapperson can you define "extreme" and which chart type you are getting lags? I've tried to get the most clean implementation, but I can easily revert to your one. I just want to understand better your extreme use cases.

Reply:
@mattapperson are you sure? 20k series with 50 data points each? beside of the low level of visual rendering a chart with 20k lines can be, I tried and I could barely render such amount of lines on my mac. Not sure if and how you have achieved it, can you send me your test code?

Reply:
I've tried with 50 series and 20k datapoints and, on my macbookpro (2,9 GHz Intel Core i7 with 4gb radeon graphics) it's lagging a bit but it's quite usable. this means drawing 50 paths with a lot of points. What you were saying is that you were drawing 20.000 path, so 20.000 dom elements that btw its a bit too much also for svg.
Could you also tell me your computer specs?

Reply:
@mattapperson I've tried to compare the two implementation, with the following configuration:
50 series * 1000 datapoint each
The example code is more or less the following for the two implementation. I've disabled for each implementation the animation, the background white border lines and the markers, and reduced the size of the line to 0.5 px:  
```
const series = new Array(50)
  .fill(0)
  .map(() => (new Array(1000).fill(0).map((d,i) => ({ x: i, y: Math.random() * 10 }))))

export default () => (
  <EuiXYChart
    width={600}
    height={200}
  >
    {
      series.map(d => {
        return <EuiLine name="Users" data={d} />
      })
    }
  </EuiXYChart>
);
```
This is the result with my current implementation
![marco_implementation mov](https://user-images.githubusercontent.com/1421091/41785658-97086840-7643-11e8-9088-559eb5caa0b2.gif)

This with the last version of your eui/chart branch
![matt_implementation mov](https://user-images.githubusercontent.com/1421091/41785965-79f4fc40-7644-11e8-91bb-f98c6e8a7470.gif)

Reply:
Interesting... I get the same result as you for the last implementation, but yours I get a very different result from (much slower). @chandlerprall would you be able to run this on your box and see what results you get?

Reply:
Just talked with Matt, he used the same macbook pro machine 2,9 GHz Intel Core i7 with 4gb radeon but tested over safari. My test were on chrome.

Reply:
With animation & line markers enabled, both are unusable at 10+ seconds between updates, locking the entire browser.

With animation & line markers disabled, my experience matches Marco's although for me both cases are _slightly_ slower.